src/System.Text.Json/tests/JsonElementWriteTests.cs

```diff
@@ -0,0 +1,926 @@
+    ""utf8"": ""p\u00cdzza"",
+    ""utf8ExtraEscape"": ""p\u00cdzza"",
+    ""arr"": [
+      ""hello"",
+      ""sailor"",
+      21,
+      ""blackjack!""
+    ],
+    ""obj"": {
+      ""arr"": [
+        1,
+        3,
+        5,
+        7,
+        11
+      ]
+    }
+  }
+}",
+                "{\"data\":" +
+                    "{\"int\":42,\"quadratic googol\":1e400,\"precisePi\":3.141592653589793238462643383279," +
+                    "\"lit0\":null,\"lit1\":false,\"lit2\":true,\"ascii\":\"pizza\",\"escaped\":\"pizza\"," +
+                    "\"utf8\":\"p\\u00cdzza\",\"utf8ExtraEscape\":\"p\\u00cdzza\"," +
+                    "\"arr\":[\"hello\",\"sailor\",21,\"blackjack!\"]," +
+                    "\"obj\":{\"arr\":[1,3,5,7,11]}}}");
+        }
+
+        [Fact]
+        public static void WriteIncredibleDepth()
+        {
+            const int TargetDepth = 500;
+            JsonReaderOptions optionsCopy = s_readerOptions;
+            optionsCopy.MaxDepth = TargetDepth + 1;
+            const int SpacesPre = 12;
+            const int SpacesSplit = 85;
+            const int SpacesPost = 4;
+
+            byte[] jsonIn = new byte[SpacesPre + TargetDepth + SpacesSplit + TargetDepth + SpacesPost];
+            jsonIn.AsSpan(0, SpacesPre).Fill((byte)' ');
+            Span<byte> openBrackets = jsonIn.AsSpan(SpacesPre, TargetDepth);
+            openBrackets.Fill((byte)'[');
+            jsonIn.AsSpan(SpacesPre + TargetDepth, SpacesSplit).Fill((byte)' ');
+            Span<byte> closeBrackets = jsonIn.AsSpan(SpacesPre + TargetDepth + SpacesSplit, TargetDepth);
+            closeBrackets.Fill((byte)']');
+            jsonIn.AsSpan(SpacesPre + TargetDepth + SpacesSplit + TargetDepth).Fill((byte)' ');
+
+            using (ArrayBufferWriter buffer = new ArrayBufferWriter(jsonIn.Length))
+            using (JsonDocument doc = JsonDocument.Parse(jsonIn, optionsCopy))
+            {
+                var writer = new Utf8JsonWriter(buffer);
+                doc.RootElement.WriteValueTo(ref writer);
+                writer.Flush();
+
+                ArraySegment<byte> formattedSegment = buffer.Formatted;
+                ReadOnlySpan<byte> formatted = formattedSegment;
+
+                Assert.Equal(TargetDepth + TargetDepth, formatted.Length);
+                Assert.True(formatted.Slice(0, TargetDepth).SequenceEqual(openBrackets), "OpenBrackets match");
+                Assert.True(formatted.Slice(TargetDepth).SequenceEqual(closeBrackets), "CloseBrackets match");
+            }
+        }
+
+        private static void WriteSimpleValue(bool indented, string jsonIn, string jsonOut = null)
+        {
+            using (ArrayBufferWriter buffer = new ArrayBufferWriter(1024))
+            using (JsonDocument doc = JsonDocument.Parse($" [  {jsonIn}  ]", s_readerOptions))
+            {
+                JsonElement target = doc.RootElement[0];
+
+                var state = new JsonWriterState(
+                    new JsonWriterOptions
```

Comment:
Is `SkipValidation` relevant to test?

Reply:
If I understand what it does (just stops things from throwing), I'd say no... aside from writing as a property not in an object context.

Indented made semantic sense to test because an easy implementation that we decided was wrong was just copying the input stream into the writer